### PR TITLE
configure.in: Modernize detection of python & boost

### PR DESCRIPTION
**Pull request by jepler**
Hopefully, this helps with autodetecting the right boost library version. It also enforces the minimum Python version of 3.4.
    
This brings us deeper into the autoconf "ecosystem" so the files config.guess et al had to be added.  Oh well!

### CODE REVIEW

**Comment by jepler** [comment]
This is green in Github Actions and Buildbot now.  It's a net growth in code, but much of the addition is maintained by others (the copies of the m4 macros and the files from autoconf) so overall it's a simplification.

File: src/configure.ac [hunk 1]
```diff
@@ -1064,28 +999,19 @@ install with "sudo apt-get install yapps2"])
 fi
 AC_SUBST(YAPPS)
 
+echo A
```
Comment (SebKuzminsky) [inline src/configure.ac:1002]: Oops.

File: src/configure.ac [hunk 2]
```diff
@@ -1458,61 +1380,48 @@ AC_MSG_CHECKING([whether to check for runtime dependencies])
...
+    else
+        AC_MSG_RESULT(no)
+        AC_MSG_ERROR([BWidget not found!
+    install with "sudo apt-get install bwidget"])
+    fi
 
-AC_MSG_CHECKING([for python pango and cairo modules])
-if $PYTHON -c 'import sys;raise SystemExit(sys.hexversion<0x3040000)'
-then
+    AC_MSG_CHECKING([for BLT using tclsh8.5])
```
Comment (SebKuzminsky) [inline src/configure.ac:1392]: Shouldn't this use `$TCLSH` too?

**Review by SebKuzminsky — CHANGES_REQUESTED** [review]
Two minor issues, but other than that thanks for the cleanup.

File: src/configure.ac (hunk 1 above)
Comment (jepler) [reply src/configure.ac]: fixed

File: src/configure.ac (hunk 2 above)
Comment (jepler) [reply src/configure.ac]: Modernized.  The reason was hidden deep in the git blame, something about ubuntu hardy :-/

**Pull request merged**